### DISCUSSION TITLE
ModuleNotFoundError: No module named 'torch_geometric.utils.to_dense_adj'

### DISCUSSION THREAD

**Discussion by iberglami** [pyg-team/pytorch_geometric#9023] [Q&A]
I am running the example of pytorch geometric temporal, the terminal pop up an error as the title. I checked the env documents and I can found the py document '_to_dense_adj.py' and the functions in it. I wonder why the programme cannot found it and raise the error.

**Comment by rusty1s** [comment]
We changed the filenames internally, but it looks like PyG Temporal requires an earlier version of PyG. At best, just import either from `torch_geometric.utils` directly or from `torch_geometric.utils._to_dense_adj`.

**Reply by Bugs-Owner** [reply]
Hi Matthias, thank you. I change "from torch_geometric.utils.to_dense_adj import to_dense_adj" to "from torch_geometric.utils import to_dense_adj", and now it's work~

**Reply by AAymanAbdelaal** [reply]
Hi. Importing dense_adj solely works with me using 'from torch_geometric.utils import to_dense_adj'. However, when I try to import any of the torch_geomteric_temporal modules such as 'from torch_geometric_temporal.dataset import METRLADatasetLoader', it fails with an error **ModuleNotFoundError: No module named 'torch_geometric.utils.to_dense_adj'**

**Reply by rusty1s** [reply]
It looks like https://github.com/benedekrozemberczki/pytorch_geometric_temporal does not yet support PyG 2.5.0. You would need to send a PR for this or manually change the imports inside the package.

**Reply by AAymanAbdelaal** [reply]
@rusty1s Thanks a lot; I will depend on the main packages instead.

**Reply by harirRzi** [reply]
Hi. After I installed torch_geometric_temporal and imported the following:

```
from torch_geometric.utils import to_dense_adj
from torch_geometric_temporal.nn.recurrent import GConvGRU
```

I get the following error:

```
from torch_geometric.utils.to_dense_adj import to_dense_adj
ModuleNotFoundError: No module named 'torch_geometric.utils.to_dense_adj'
```

I appreciate if you help me how can I fix it?

**Reply by rusty1s** [reply]
It looks like PyG Temporal is not yet compatible with PyG 2.5.0. You can manually change the import from
```
from torch_geometric.utils.to_dense_adj import to_dense_adj
```
to
```
from torch_geometric.utils import to_dense_adj
```

**Comment by dipanshuhaldar** [comment]
Hi I have trying to access torch geometric temporal but the imports are giving me errors

import torch
import numpy as np
from torch_geometric_temporal.signal import temporal_signal_split
from torch_geometric_temporal.signal import StaticGraphTemporalSignal
from torch_geometric_temporal.nn.recurrent import GConvGRU
import torch.nn.functional as F

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[7], line 3
      1 import torch
      2 import numpy as np
----> 3 from torch_geometric_temporal.signal import temporal_signal_split
      4 from torch_geometric_temporal.signal import StaticGraphTemporalSignal
      5 from torch_geometric_temporal.nn.recurrent import GConvGRU

File [/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/__init__.py:1](https://vsjxvzhe42klhpn.studio.us-east-1.sagemaker.aws/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/__init__.py#line=0)
----> 1 from torch_geometric_temporal.nn import *
      2 from torch_geometric_temporal.dataset import *
      3 from torch_geometric_temporal.signal import *

File [/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/nn/__init__.py:2](https://vsjxvzhe42klhpn.studio.us-east-1.sagemaker.aws/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/nn/__init__.py#line=1)
      1 from .recurrent import *
----> 2 from .attention import *
      3 from .hetero import *

File [/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/nn/attention/__init__.py:6](https://vsjxvzhe42klhpn.studio.us-east-1.sagemaker.aws/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/nn/attention/__init__.py#line=5)
      4 from .gman import GMAN, SpatioTemporalEmbedding, SpatioTemporalAttention
      5 from .mtgnn import MTGNN, MixProp, GraphConstructor
----> 6 from .tsagcn import GraphAAGCN, AAGCN
      7 from .dnntsp import DNNTSP

File [/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/nn/attention/tsagcn.py:6](https://vsjxvzhe42klhpn.studio.us-east-1.sagemaker.aws/opt/conda/lib/python3.11/site-packages/torch_geometric_temporal/nn/attention/tsagcn.py#line=5)
      4 import torch.nn as nn
      5 from torch.autograd import Variable
----> 6 from torch_geometric.utils.to_dense_adj import to_dense_adj
      7 import torch.nn.functional as F
     10 class GraphAAGCN:

ModuleNotFoundError: No module named 'torch_geometric.utils.to_dense_adj'

**Reply by dipanshuhaldar** [reply]
from torch_geometric.utils import to_dense_adj does not help either.
